src/components/Ide.js

```diff
@@ -1,19 +1,42 @@
-import React, { useState } from 'react';
-import CodeMirror from '@uiw/react-codemirror';
-import 'react-codemirror/node_modules/codemirror/mode/javascript/javascript';
-import 'react-codemirror/node_modules/codemirror/mode/css/css';
-import 'react-codemirror/node_modules/codemirror/addon/hint/show-hint';
-import 'react-codemirror/node_modules/codemirror/addon/hint/javascript-hint';
-import 'react-codemirror/node_modules/codemirror/theme/monokai.css';
+import React, { useState, useEffect } from "react";
+import CodeMirror from "@uiw/react-codemirror";
+import "react-codemirror/node_modules/codemirror/mode/javascript/javascript";
+import "react-codemirror/node_modules/codemirror/mode/css/css";
+import "react-codemirror/node_modules/codemirror/addon/hint/show-hint";
+import "react-codemirror/node_modules/codemirror/addon/hint/javascript-hint";
+import "react-codemirror/node_modules/codemirror/theme/monokai.css";
+
+import { javascript } from "@codemirror/lang-javascript";
+import { autocompletion } from "@codemirror/autocomplete";
 const languageModes = [
-  { name: 'javascript', json: true },
-  { name: 'python', version: 3 },
-  { name: 'ruby' },
+  { name: "javascript", json: true },
+  { name: "python", version: 3 },
+  { name: "ruby" },
 ];
 
+//To be refactored & changed
+function myCompletions(context) {
+  let word = context.matchBefore(/\w*/);
+  if (word.from === word.to && !context.explicit) return null;
+
+  const doc = context.state.doc;
+  const pos = word.from;
+
+  const jsState = javascript.languageData
+    .of({})
```

Comment:
![image](https://github.com/akhil27styles/Code-Collab/assets/52193850/c4de69bb-275b-4e50-a9f9-5f6222d07823)
getting this error here